### ISSUE TITLE
Feature request: Recursive multistep multivariate forecasting and direct multistep multi-series forecasting

### ISSUE THREAD

**Issue by KishManani** [JoaquinAmatRodrigo/skforecast#625]
Currently `ForecastMultiSeriesCustom` and `ForecastMultiSeries` implement a recursive approach to multi-series forecasting. `ForecastMultiVariate` implements a direct forecasting approach for multivariate forecasting.

It would be good to have:

- Direct forecasting available for multi-series forecasting.
- Recursive forecasting available for multivariate forecasting.
- A consistent naming approach to which methods use a direct or recursive approach. Some classes use a recursive approach as default (e.g., `ForecasterAutoreg`, `ForecasterMultiseries`) and a direct approach is specified in the class name (i.e., `ForecasterAutoregDirect`). However, for multivariate forecasting, a direct approach is used but not reflected in the class name (i.e., `ForecasterMultivariate`). 

Thanks again!
Kishan

**Comment by JavierEscobarOrtiz** [comment]
Hello Kishan,

Thanks for your comments,

> Direct forecasting available for multi-series forecasting.

This is something we have pending. Since we are still developing new features for `ForecasterMultiseries`, we will wait for a more stable version before implementing it. 

> Recursive forecasting available for multivariate forecasting.

As it is now, the `ForecasterAutoregMultiVariate` is only trained in one level (series 1 or series 2) and creates one model per step (direct approach). In the figure, let's say that if you specify `level = 'Series 1'` when creating the Forecaster, it will only use the training matrix at the top.

![forecaster_multivariate_train_matrix_diagram](https://github.com/JoaquinAmatRodrigo/skforecast/assets/95030312/dbf1231b-178f-4939-9695-2b754e492d21)

Since this forecaster needs the future values of the series to predict the following steps, it needs 2 models (one for each series) to make it recursive and call them in parallel. Predict step 1 for both series, predict step 2 for both series...

Is that what you have in mind?

> A consistent naming approach

Completly agree. We have this figure in the repo page and documentation to help the users with this problem:

![imagen](https://github.com/JoaquinAmatRodrigo/skforecast/assets/95030312/af35ce95-874e-4d82-a31d-23154f22dc87)

https://skforecast.org/latest/#forecasters
https://github.com/JoaquinAmatRodrigo/skforecast#forecasters

Best,

Javi

**Comment by JoaquinAmatRodrigo** [comment]
The names of the Forecasters definitely need to be revisited, the family is growing and at some point we need to make the naming more intuitive. This may require some proper alias management to avoid breaking compatibility. ¡Maybe one day we are ready for version 1.0.0. ! :smile: :smile:

**Comment by JoaquinAmatRodrigo** [comment]
**Recursive forecasting available for multivariate forecasting**

This is quite challenging because this hybrid strategy requires each individual series to be predicted internally so that the predicted values can then be used as input features by a second model. We have skipped this so far because the idea of this approach being too computationally expensive, but we could do some experimentation to test its feasibility.

**Comment by KishManani** [comment]
> **Recursive forecasting available for multivariate forecasting**
> 
> This is quite challenging because this hybrid strategy requires each individual series to be predicted internally so that the predicted values can then be used as input features by a second model. We have skipped this so far because the idea of this approach being too computationally expensive, but we could do some experimentation to test its feasibility.

Yes I agree. This would be computationally challenging. I think there are other features which are easier to add to `skforecast` which would add more value.

**Comment by JoaquinAmatRodrigo** [comment]
Hi everyone! 👋

It’s been a while since the last activity on this issue, so I’m going to close it to help us keep our backlog manageable and focused on active tasks.

If you are still experiencing this problem or believe this feature is still highly relevant for the latest version, please feel free to leave a comment or reopen the issue. Thank you for your contribution!
